polarization/jonesvector.py

```diff
@@ -126,10 +132,10 @@ def __rmul__(self, leftSide):
 
     def mul_number(self, n):
         """ Multiply a Jones vector by a number."""
-        return JonesVector(Ex=self.Ex * n, Ey=self.Ey * n, k=self.k, z=self.z)
+        return JonesVector(Ex=self.E1 * n, Ey=self.E2 * n, k=self.k, z=self.z)
```

Comment:
This is wrong: we need to use `self.Ex`, because `self.b1` may have been rotated.

Reply:
Elsewhere in the code, we access E1 and E2 directly because we work under the assumption that the `JonesMatrix` and the `JonesVector` are on the same basis (this is fine for now, because we know we are not changing `b1` and `b2`, but may not always be true.